### PR TITLE
Add doc for spacemacs/open-junk-file keybinding

### LINKED ISSUE #4398: `SPC f J` should be documented in doc/DOCUMENTATION.org
`SPC f J` (`spacemacs/helm-open-junk-file`) is a very useful command.  Although I often use the `*scratch*` buffer (`SPC b s`) for things like testing programming language features, sometimes one buffer isn't enough.

I think this command should be documented in `doc/DOCUMENTATION.org`.

### PR DESCRIPTION
**Pull request by balajisivaraman**
Closes #4398

### CODE REVIEW

File: doc/DOCUMENTATION.org [hunk 1]
```diff
@@ -1619,29 +1619,30 @@ Spacemacs marks buffers as useful. For instructions, see the [[file:FAQ.org::Cha
...
+| ~SPC f b~   | go to file bookmarks                                             |
+| ~SPC f c~   | copy current file to a different location                        |
+| ~SPC f C d~ | convert file from unix to dos encoding                           |
+| ~SPC f C u~ | convert file from dos to unix encoding                           |
+| ~SPC f D~   | delete a file and the associated buffer (ask for confirmation)   |
+| ~SPC f E~   | open a file with elevated privileges (sudo edit)                 |
+| ~SPC f f~   | open file with =helm= (or =ido=)                                 |
+| ~SPC f F~   | try to open the file under point =helm=                          |
+| ~SPC f j~   | jump to the current buffer file in dired                         |
+| ~SPC f J~   | open a junk file (in =fundamental mode=) using =helm= (or =ivy=) |
```
Comment (nixmaniack) [inline doc/DOCUMENTATION.org]: Not necessarily `fundamental-mode`, it depends on the extension you provide.

Comment (balajisivaraman) [inline doc/DOCUMENTATION.org]: @nixmaniack, Thank you! Did not know that, will update the PR accordingly.

**Comment by robbyoconnor** [comment]
:+1:

**Comment by syl20bnr** [comment]
Thank you again :-)
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**